tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -296,10 +296,8 @@ internal static IDogStatsd CreateDogStatsdClient(ImmutableTracerSettings setting
                 switch (settings.ExporterInternal.MetricsTransport)
                 {
                     case MetricsTransportType.NamedPipe:
-                        // Environment variables for windows named pipes are not explicitly passed to statsd.
-                        // They are retrieved within the vendored code, so there is nothing to pass.
-                        // Passing anything through StatsdConfig may cause bugs when windows named pipes should be used.
                         Log.Information("Using windows named pipes for metrics transport.");
+                        config.PipeName = settings.ExporterInternal.MetricsPipeNameInternal;
```

Comment:
Just want to double check, but the `MetricPipeNameInternal` is `null` by default (so in the config key), and it could also be `null` here -> https://github.com/DataDog/dd-trace-dotnet/blob/f41a0d82b2f128303cf8aecdcf84cb2235877774/tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs#L293

So would we need to check to make sure it has a value here before setting it to `config.PipeName`?

Reply:
I don't _think_ so. The only way for `MetricsTransport` to be `NamedPipes` is if the value is non-null, and we set the pipe name there:

https://github.com/DataDog/dd-trace-dotnet/blob/f41a0d82b2f128303cf8aecdcf84cb2235877774/tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs#L315-L316

Of course, worst case, it _is_ null somehow, in which case this whole PR is a no-op 😃

Reply:
Ahh yep makes sense!